datacats/environment.py

```diff
@@ -130,20 +178,106 @@ def new(cls, path, ckan_version, port=None):
                 ' does not exist')
 
         datadir = expanduser('~/.datacats/' + name)
-        target = workdir + '/' + name
-
+        childdir = join(datadir, child_name)
+        # We track through the datadir to the target if we are just making a
+        # child
         if isdir(datadir):
-            raise DatacatsError('Environment data directory {0} already exists',
-                (datadir,))
-        if isdir(target):
-            raise DatacatsError('Environment directory already exists')
-
-        environment = cls(name, target, datadir, ckan_version, port)
+            with open(join(datadir, 'project-dir')) as pd:
+                target = pd.read()
+        else:
+            target = workdir + '/' + name
+
+        if isdir(childdir):
+            raise DatacatsError('Child environment data directory {0} already exists',
+                (childdir,))
+        # This is the case where the data dir has been removed,
+        if isdir(target) and not isdir(datadir):
+            raise DatacatsError('Environment directory exists, but data directory does not.\n'
+                                'If you simply want to recreate the data directory, run "datacats init"'
+                                ' in the environment directory.')
+
+        environment = cls(name, target, datadir, child_name, ckan_version, port)
         environment._generate_passwords()
         return environment
 
     @classmethod
-    def load(cls, environment_name=None, data_only=False):
+    def _needs_format_conversion(cls, datadir):
+        """
+        Returns True if `datadir` requires conversion to the new format (child envs).
+        """
+        return isdir(datadir) and isdir(join(datadir, 'run')) and exists(join(datadir, 'passwords.ini')) and exists(
+                join(datadir, 'search')) and exists(join(datadir, 'solr')) and exists(join(datadir, 'project-dir'))
+
+    @classmethod
+    def _convert_environment(cls, datadir):
+        new_child_name = 'primary'
+        inp = None
+
+        while inp != 'y' and inp != 'n':
+            inp = raw_input('You are using a file in the old DataCats format. Would you like to convert it (y/n) [n]: ')
```

Comment:
How about a `datacats migrate` command instead. We can have the other commands fail on load() with an error saying "this environment is out of date, please run "datacats migrate" to upgrade it to the latest version."